x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/configuration_step/analysis_fields_table.tsx

```diff
@@ -67,141 +69,141 @@ const columns = [
-            defaultMessage: 'Excluded fields',
+  return (
+    <Fragment>
+      <EuiFormRow
+        label={i18n.translate('xpack.ml.dataframe.analytics.create.includedFieldsLabel', {
+          defaultMessage: 'Included fields',
+        })}
+      >
+        <Fragment />
+      </EuiFormRow>
+      {tableItems.length > 0 && (
+        <EuiText size="xs">
+          {includes.length}
+          {i18n.translate('xpack.ml.dataframe.analytics.create.includedFieldsCount', {
+            defaultMessage: ' fields included in the analysis',
+          })}
+        </EuiText>
+      )}
+      {tableItems.length === 0 && (
+        <EuiCallOut
+          title={i18n.translate('xpack.ml.dataframe.analytics.create.calloutTitle', {
+            defaultMessage: 'Analysis fields not available',
           })}
-          helpText={i18n.translate(
-            'xpack.ml.dataframe.analytics.create.excludedFieldsLabelHelpText',
-            {
-              defaultMessage: 'From included fields, select fields to exclude from analysis.',
-            }
-          )}
         >
-          <Fragment />
-        </EuiFormRow>
-        {tableItems.length === 0 && (
-          <EuiCallOut
-            title={i18n.translate('xpack.ml.dataframe.analytics.create.calloutTitle', {
-              defaultMessage: 'Analysis fields not available',
-            })}
-          >
-            <FormattedMessage
-              id="xpack.ml.dataframe.analytics.create.calloutMessage"
-              defaultMessage="Additional data required to load analysis fields."
-            />
-          </EuiCallOut>
-        )}
-        {tableItems.length > 0 && (
-          <EuiPanel paddingSize="m" data-test-subj="mlAnalyticsCreateJobWizardExcludesSelect">
-            <CustomSelectionTable
-              data-test-subj="mlAnalyticsCreationAnalysisFieldsTable"
-              checkboxDisabledCheck={checkboxDisabledCheck}
-              columns={columns}
-              filters={filters}
-              items={tableItems}
-              itemsPerPage={5}
-              onTableChange={(selection: FieldSelectionItem[]) => {
-                setCurrentSelection(selection);
-              }}
-              selectedIds={currentSelection}
-              singleSelection={false}
-              sortableProperties={sortableProperties}
-              tableItemId={'name'}
-            />
-          </EuiPanel>
-        )}
-        <EuiSpacer />
-      </Fragment>
-    );
-  },
-  (prevProps, nextProps) => prevProps.tableItems.length === nextProps.tableItems.length
-);
+          <FormattedMessage
+            id="xpack.ml.dataframe.analytics.create.calloutMessage"
+            defaultMessage="Additional data required to load analysis fields."
+          />
+        </EuiCallOut>
+      )}
+      {tableItems.length > 0 && (
+        <EuiPanel paddingSize="m" data-test-subj="mlAnalyticsCreateJobWizardIncludesSelect">
+          <CustomSelectionTable
```

Comment:
I see weird behavior when I try and use the 'select / deselect all' checkbox. If I try to deselect all, on next render all the checkboxes become selected again. I guess it's because the explain API doesn't accept the config where all fields are excluded, so the return marks all fields as included? Looks like the 'deselect all' action is deselecting the dependent field, so a fix might be to make sure the dependent field retains its selection, which would then  leave the user in a state where they need to select at least one other field for inclusion?

The page size also reverts to the default of 5:

![includes_all](https://user-images.githubusercontent.com/7405507/86120663-9e4a8500-bacc-11ea-937f-fb42bf52f960.gif)

Reply:
This has been fixed in 25f859e29c59017dd8b1d4e7085b2a1cd7a3de22

For jobs with dependent_variable - we now make sure dependent_variable is always in the includes selection. For outlier detection we show a minimal fields required message if `includes` is empty.